Source/Fuse.PushNotifications/Android/Impl.uno

```diff
@@ -296,13 +296,15 @@ namespace Fuse.PushNotifications
 		[Foreign(Language.Java)]
 		static void SpitOutNotification(Java.Object _listener, string title, string body, string bigTitle, string bigBody, string notificationStyle, string featuredImage, string sound, Java.Object _payload)
 		@{
+	        int requestID = (int) System.currentTimeMillis();
 			Context context = (Context)_listener;
 			Bundle payload = (Bundle)_payload;
 			Intent intent = new Intent(context, @(Activity.Package).@(Activity.Name).class);
-			intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        	intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
 			intent.setAction(PushNotificationReceiver.ACTION);
 			intent.replaceExtras(payload);
-			android.app.PendingIntent pendingIntent = android.app.PendingIntent.getActivity(context, 0, intent, android.app.PendingIntent.FLAG_ONE_SHOT);
+
+			android.app.PendingIntent pendingIntent = android.app.PendingIntent.getActivity(context, requestID, intent, android.app.PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
Excuse me if I'm being stupid here, but let me check if I'm understanding this correctly...

requestCode = 0 is replaced with System.currentTimeMillis(), to make each request unique, and then replace `FLAG_ONE_SHOT` is replaced with `FLAG_UPDATE_CURRENT` to deal with the off-chance of two push-notifications triggering within the same millisecond (or whatever granularity System.currentTimeMillis() has)?

I think a short explanation about the intent of the changes would make sense to put in the commit-message.

Reply:
@devadiab I'd prefer using `PushNotificationReceiver.nextID()` here.

Reply:
@devadiab if possible it would also be great to hear why we had to change the `FLAG_UPDATE_CURRENT` flag. As @kusma says having that in the commit message would be great. Android is such a sea of weirdness that trying to understand the logic of changes like this gets really hard 8 months down the line :)

Reply:
@kusma , @cbaggers  Thanks guys for you follow-up. about the Id i was thinking to return it to `PushNotificationReceiver.nextID()` .
About the  `FLAG_UPDATE_CURRENT` flag, android says: 
> Flag indicating that if the described PendingIntent already exists, then keep it but replace its extra data with what is in this new Intent. 
which mean that it will let the app received the new payload and extras.